### PR TITLE
#449 Monospace font in search and replace combobox

### PR DESCRIPTION
**Pull request by tonyk24**

### CODE REVIEW

**Comment by tonyk24** [comment]
Before and after:

after
before

File: PowerEditor/src/ScitillaComponent/FindReplaceDlg.cpp [hunk 1]
```diff
@@ -681,6 +681,15 @@ INT_PTR CALLBACK FindReplaceDlg::run_dlgProc(UINT message, WPARAM wParam, LPARAM
 	{
 		case WM_INITDIALOG :
 		{
+			HWND hFindCombo = ::GetDlgItem(_hSelf, IDFINDWHAT);
+			HWND hReplaceCombo = ::GetDlgItem(_hSelf, IDREPLACEWITH);
+			HWND hFiltersCombo = ::GetDlgItem(_hSelf, IDD_FINDINFILES_FILTERS_COMBO);
+
+			HFONT hf = CreateFont(-12, 0, 0, 0, 0, FALSE, 0, 0, 0, 0, 0, 0, 0, _T("Courier New"));
```
Comment (SinghRajenM) [inline PowerEditor/src/ScitillaComponent/FindReplaceDlg.cpp]: Somewhere ```DeleteObject``` should be called, shouldn't?

Comment (tonyk24) [inline PowerEditor/src/ScitillaComponent/FindReplaceDlg.cpp]: True that.

Accidently made change to a new pull request:

https://github.com/notepad-plus-plus/notepad-plus-plus/pull/3361

**Comment by Ede123** [comment]
I'm afraid I don't like it:
* it unnecessarily blows up the size of the text boxes
* it's bad for the overall visual appearance of the dialog
* most importantly: it allows for a lot less characters to be visible when I'm already often using more than the full size of the input which reduces efficiency a lot

The only reason where a mono-spaced font would be useful is when opening the dropdown, but this is a very minor improvement an in my opinion it does not outweigh the downsides by far.

A useful change would the better visibility of upper case "i" vs lower case "l" but for that there are much better options (e.g. usage of [Source Sans Pro](https://fonts.google.com/specimen/Source+Sans+Pro) or similar)

**Comment by sasumner** [comment]
With the relatively new (I think 7.4.2 and beyond) ability to resize the Find dialog window, I think I like this!  Without that ability I would have to agree with @Ede123 's "most importantly" comment.

**Comment by donho** [comment]
Let's keep the current behaviour.

**Comment by sasumner** [comment]
Wow, this is so old I'd forgotten about it.  VERY coincidentally, just yesterday I implemented a Pythonscript solution so that I can have this behavior with any Notepad++ version.  Yay for me! 

image

**Pull request closed without merging**